### ISSUE TITLE
Npcap 1.70 & 1.71 fails to uninstall

### ISSUE THREAD

**Issue by gbloice** [nmap/npcap#610]
Currently have 1.60 installed, Win 10 21H2 up to date.

From the uninstaller:

```
Reading service options from registry
Output folder: C:\Program Files\Npcap
Rename: C:\WINDOWS\system32\Npcap\wpcap.dll->C:\WINDOWS\system32\Npcap\wpcap.dll.del
Rename: C:\WINDOWS\system32\Npcap\Packet.dll->C:\WINDOWS\system32\Npcap\Packet.dll.del
Rename: C:\WINDOWS\system32\Npcap\NpcapHelper.exe->C:\WINDOWS\system32\Npcap\NpcapHelper.exe.del
Rename: C:\WINDOWS\system32\Npcap\WlanHelper.exe->C:\WINDOWS\system32\Npcap\WlanHelper.exe.del
Rename: C:\WINDOWS\system32\Npcap\wpcap.dll->C:\WINDOWS\system32\Npcap\wpcap.dll.del
Rename: C:\WINDOWS\system32\Npcap\Packet.dll->C:\WINDOWS\system32\Npcap\Packet.dll.del
Rename: C:\WINDOWS\system32\Npcap\NpcapHelper.exe->C:\WINDOWS\system32\Npcap\NpcapHelper.exe.del
Rename: C:\WINDOWS\system32\Npcap\WlanHelper.exe->C:\WINDOWS\system32\Npcap\WlanHelper.exe.del
Stopping the npcap driver
Unable to stop driver.
Rename: C:\WINDOWS\system32\Npcap\wpcap.dll.del->C:\WINDOWS\system32\Npcap\wpcap.dll
Rename: C:\WINDOWS\system32\Npcap\Packet.dll.del->C:\WINDOWS\system32\Npcap\Packet.dll
Rename: C:\WINDOWS\system32\Npcap\NpcapHelper.exe.del->C:\WINDOWS\system32\Npcap\NpcapHelper.exe
Rename: C:\WINDOWS\system32\Npcap\WlanHelper.exe.del->C:\WINDOWS\system32\Npcap\WlanHelper.exe
Rename: C:\WINDOWS\system32\Npcap\wpcap.dll.del->C:\WINDOWS\system32\Npcap\wpcap.dll
Rename: C:\WINDOWS\system32\Npcap\Packet.dll.del->C:\WINDOWS\system32\Npcap\Packet.dll
Rename: C:\WINDOWS\system32\Npcap\NpcapHelper.exe.del->C:\WINDOWS\system32\Npcap\NpcapHelper.exe
Rename: C:\WINDOWS\system32\Npcap\WlanHelper.exe.del->C:\WINDOWS\system32\Npcap\WlanHelper.exe
Unable to stop driver. Npcap service is 'Write-Host : The 'Write-Host' command was found in the module 'Boxstarter.Chocolatey', but the module could not be 
loaded. For more information, run 'Import-Module Boxstarter.Chocolatey'.
At line:1 char:41
+ Stop-Service -Name npcap -PassThru | %{ Write-Host $_.Status }
+                                         ~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Write-Host:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CouldNotAutoloadMatchingModule
 
'. Uninstall reverted.
Unable to stop driver
```

I've had boxstarter installed on this machine for almost 3 years.

I normally run PowerShell 7, I also attempted a command line uninstall from an elevated PowerShell 5 prompt but got the same results.

I suspect this is one of those annoying PowerShell 5\7 incompatibility issues.

Finally got uninstall to complete from a PowerShell7 elevated prompt by first running `Import-Module Boxstarter.Chocolatey` to pre-load the module causing issues.

**Comment by rknall** [comment]
Do you guys need more information on fixing this? Would like to update the Wireshark builder for 1.70 and holding off atm.

**Comment by gbloice** [comment]
Issue still present with 1.71, the workaround I noted above still works (install from a a PowerShell7 elevated prompt by first running `Import-Module Boxstarter.Chocolatey` to pre-load the module causing issues, then run npcap-1.71.exe

**Comment by dmiller-nmap** [comment]
@gbloice Can you clarify whether the issue is present when trying to uninstall Npcap 1.71 or when trying to upgrade to 1.71 from an earlier version? Can you provide the install.log from uninstalling Npcap 1.71?

**Comment by guyharris** [comment]
> Can you clarify whether the issue is present when trying to uninstall Npcap 1.71 or when trying to upgrade to 1.71 from an earlier version?

I.e., it sounds as if uninstalling 1.60, as part of the process of installing 1.70 or 1.71, is what's failing.

**Comment by gbloice** [comment]
Sorry for the delay in responding.

The issue came up when upgrading 1.60 to 1.70 and again when upgrading 1.70 to 1.71.

The issue seemed to me to be a clash of external PS cmdlets such that the uninstall script failed to run.

**Comment by Walter784** [comment]
I just upgraded from 1.60 to 1.70 on Win10 64-bit machine and had same problem. Initially unable to unstall Npcap either via Wireshark upgrade or via Windows Application uninstall.
But I rebooted my machine and was finally able to properly uninstall Npcap 1.60 and then successfully upgrade to 1.70.

Cheers,

**Comment by vp-trellix** [comment]
I am unable to uninstall npcap with 1.7x versions. Isnt there a cleanup utility?

**Comment by Walter784** [comment]
I performed a reboot, followed by opening up Windows application manager and manually uninstalling it from Windows. I tried it before rebooting the PC but it wouldn't work. But after rebooting the PC, it finally worked. 
Then you can re-install Wireshark with no problem.

**Comment by fyodor** [comment]
Hi all.  I'm closing this as we haven't heard anything more about the issue since 2022.  Maybe @dmiller-nmap's improved uninstaller (which was included one newer Npcap's released since then) resolved it, or maybe BoxInstaller made a change which prevented the interference.  If anyone is still experiencing the problem, please leave the details here so we can re-open or just create a new ticket.  Thanks!

**Comment by ach-raf** [comment]
> Hi all. I'm closing this as we haven't heard anything more about the issue since 2022. Maybe @dmiller-nmap's improved uninstaller (which was included one newer Npcap's released since then) resolved it, or maybe BoxInstaller made a change which prevented the interference. If anyone is still experiencing the problem, please leave the details here so we can re-open or just create a new ticket. Thanks!

i have npcap 1.60 and i can't delete this monstrosity, why is it such a problem to remove this program?

**Comment by AquaeAtrae** [comment]
**SOLUTION**
I encountered the same problem where I had used Boxstarter to install Wireguard (v 4.0.0 perhaps) which in turn install Npcap OEM 1.6.0. Now years later I've upgraded Wireshark but found I was completely unable to uninstall Npcap. I tried everything I could think up including disabling Npcap from powershell, deleting Npcap from the \Windows\System32 and \Windows\SysWow64 directories, rebooting, safe mode. No matter what, the uninstaller kept trying to use an overwritten Write-Host command to stop the service (which no longer existed). I know Boxstarter uses a modified version of the Write-Host command to log its build process and apparently this was hardcoded into Wireshark's uninstall processes for Npcap. 

1. Upon further investigation, I found that Boxstarter would no longer launch so I downloaded and installed v3.0.3 fresh using Chocolatey. I could then launch Boxstarter and provide my password just as I had when I used it to build this machine. 
2. I confirmed that I was now running Powershell within Boxstarter's context. It should not be using the Powershell version. 
```
Get-Command Write-Host | ft -AutoSize

CommandType Name       Version Source
----------- ----       ------- ------
Alias       Write-Host 3.0.3.0 boxstarter.chocolatey
```
3. I then located the Npcap install directory with the Uninstall.exe tool. ```cd 'C:\Program Files\Npcap\'```
4. I ran the Uninstall.exe tool from within the Boxstarter context. ```.\Uninstall.exe``` 
5. Watching the progress, I finally saw it successfully stop the Npcap service and Completed. 
6. I will now reboot and reinstall fresh copies of Wireshark using Uniget (aka Winget) and then Npcap OEM from within it's license. Hopefully similar complications won't arise in the future.  Note: Npcap's license [prevents installing it directly with package managers](https://github.com/nmap/npcap/issues/588) unfortunately.
